From 196fb3a5bddff173e53b5ecf8af141db8ef4fa9e Mon Sep 17 00:00:00 2001
From: Prakash Brahmajyosyula <bprakash@marvell.com>
Date: Wed, 20 Feb 2019 19:23:21 +0530
Subject: [PATCH 1023/1051] net: usb: asix88179_178a: fix truesize adjustments.

This patch stops changing truesize for ax88179_178a based
Gigabit Ethernet Adapter.
It fixes below WARN_ON_ONCE stack dump issue.

[<ffff000008c092c0>] skb_try_coalesce+0x370/0x378
[<ffff000008c745d8>] tcp_try_coalesce+0x68/0x180
[<ffff000008c74754>] tcp_queue_rcv+0x64/0x160
[<ffff000008c7b0fc>] tcp_rcv_established+0x43c/0x640
[<ffff000008c84984>] tcp_v4_do_rcv+0x104/0x248
[<ffff000008c8730c>] tcp_v4_rcv+0xb0c/0xc08
[<ffff000008c5f534>] ip_local_deliver_finish+0x74/0x260
[<ffff000008c5fc5c>] ip_local_deliver+0x4c/0xf0
[<ffff000008c5f8e4>] ip_rcv_finish+0x1c4/0x370
[<ffff000008c5ff80>] ip_rcv+0x280/0x3c0
[<ffff000008c1d3cc>] __netif_receive_skb_core+0x384/0x990
[<ffff000008c1d9ec>] __netif_receive_skb+0x14/0x60
[<ffff000008c1ef9c>] process_backlog+0x9c/0x158
[<ffff000008c22b94>] net_rx_action+0xf4/0x2c0
[<ffff00000808193c>] __do_softirq+0x12c/0x228
[<ffff0000080da608>] irq_exit+0x138/0x158
[<ffff000008128668>] __handle_domain_irq+0x70/0x128
[<ffff0000080816fc>] gic_handle_irq+0x7c/0x184

Change-Id: Ibc8ddb518bcd11f8c44322b830693e417df5bdbc
Signed-off-by: Prakash Brahmajyosyula <bprakash@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/4432
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/usb/ax88179_178a.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/net/usb/ax88179_178a.c b/drivers/net/usb/ax88179_178a.c
index 2207f7a7d1ff..36ddae4b2afd 100644
--- a/drivers/net/usb/ax88179_178a.c
+++ b/drivers/net/usb/ax88179_178a.c
@@ -1404,7 +1404,6 @@ static int ax88179_rx_fixup(struct usbnet *dev, struct sk_buff *skb)
 			skb_pull(skb, 2);
 			skb->len = pkt_len;
 			skb_set_tail_pointer(skb, pkt_len);
-			skb->truesize = pkt_len + sizeof(struct sk_buff);
 			ax88179_rx_checksum(skb, pkt_hdr);
 			return 1;
 		}
@@ -1414,7 +1413,6 @@ static int ax88179_rx_fixup(struct usbnet *dev, struct sk_buff *skb)
 			ax_skb->len = pkt_len;
 			ax_skb->data = skb->data + 2;
 			skb_set_tail_pointer(ax_skb, pkt_len);
-			ax_skb->truesize = pkt_len + sizeof(struct sk_buff);
 			ax88179_rx_checksum(ax_skb, pkt_hdr);
 			usbnet_skb_return(dev, ax_skb);
 		} else {
-- 
2.17.1

